fiftyone/utils/sam3.py

```
@@ -545,3 +551,740 @@ def _forward_pass_auto(self, imgs):
+        _frame_indices = (
+            frame_indices
+            if frame_indices
+            else range(1, len(sample.frames) + 1)
+        )
+
+        prompts = {}
+        for frame_number in _frame_indices:
+            if frame_number > len(sample.frames):
+                raise ValueError(
+                    f"Frame index {frame_number} not valid for {len(sample.frames)} sample frames."
+                )
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)